src/store/Network.ts

```diff
@@ -255,15 +255,18 @@ export default {
             const currentProfile: ProfileModel = rootGetters['profile/currentProfile'];
             const networkService = new NetworkService();
             const nodeService = new NodeService();
+            if (!newCandidate) {
```

Comment:
This will not fix the root issue, right? no matter it is new or not it should retry ->. timeout -> switch. 

also instead of randomly pick, it would be better to build a random node list to try 1 by 1 after certain seconds elapse. So the user get to connect to the best node